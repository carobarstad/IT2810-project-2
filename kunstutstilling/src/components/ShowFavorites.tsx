import React, { Component } from 'react'

interface State{
    active: string
    message: string
}

interface Props{
    active: string
}

export default class ShowFavorites extends Component<Props, State> {
    constructor(initState: {active:string}){
        super(initState)
        this.state = {active: 'false', message: 'Vis favoritter'};
    }

    toggleFavorites(){
        let ArtworkBoxes = document.getElementsByClassName('ArtworkBox') as HTMLCollectionOf<HTMLElement>
        if(this.state.active === 'true'){ //Already showing favorites and wants to view the rest:
            for(let i = 1; i <= ArtworkBoxes.length; i++){
                sessionStorage.setItem('liked'+i+'display', 'display: block;')
                ArtworkBoxes[i-1].setAttribute('style', sessionStorage.getItem('liked'+i+'display')! + sessionStorage.getItem('artwork'+i+'grid')!)
            }
            this.setState({active: 'false', message: 'Vis favoritter'})
        } else {
            for(let i = 1; i <= ArtworkBoxes.length; i++){
                if(!localStorage.getItem('artwork'+i)){
                    sessionStorage.setItem('liked'+i+'display', 'display: none;')
                    ArtworkBoxes[i-1].setAttribute('style', sessionStorage.getItem('liked'+i+'display')! + sessionStorage.getItem('artwork'+i+'grid')!)
                }
            }
            this.setState({active: 'true', message: 'Vis alle'})
        }
    }

    //Bruk useLayoutEffect() her?
    /*componentDidMount(){
        if(sessionStorage.getItem('favoriteDisplay')){
            console.log('fant favorites i sessionstorage')
            let ArtworkBoxes = document.getElementsByClassName('ArtworkBox') as HTMLCollectionOf<HTMLElement>
            for(let i = 0; i < ArtworkBoxes.length; i++){
                if(!localStorage.getItem('artwork'+(i+1))){
                    ArtworkBoxes[i].setAttribute('style', 'display:none')
                }
            }
            this.setState({active: 'true', message: 'Vis alle'})
        }
    }*/
    
    render() {
        return (
            <button className="SettingsButton" onClick={()=>this.toggleFavorites()}>
                {this.state.message}
            </button>
        )
    }
}
