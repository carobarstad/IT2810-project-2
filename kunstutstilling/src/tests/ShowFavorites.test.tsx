import React from 'react';
import ShowFavorites from '../components/ShowFavorites';
const ReactTestRenderer = require('react-test-renderer');

describe('Button', () => {
  it('matches snapshot', () => {
    const tree = ReactTestRenderer.create(
      <ShowFavorites active="true"/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});