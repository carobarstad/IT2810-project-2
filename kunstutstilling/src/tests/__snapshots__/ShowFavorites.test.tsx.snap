// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<button
  className="ShowFavoriteButton"
  onClick={[Function]}
>
  Vis favoritter
</button>
`;
